src/components/tx/SignOrExecuteForm/SignerForm/index.tsx

```diff
@@ -0,0 +1,103 @@
+import {
+  Box,
+  FormControl,
+  InputLabel,
+  MenuItem,
+  Select,
+  SvgIcon,
+  Tooltip,
+  Typography,
+  type SelectChangeEvent,
+} from '@mui/material'
+import { useNestedSafeOwners } from '@/hooks/useNestedSafeOwners'
+import useWallet from '@/hooks/wallets/useWallet'
+import EthHashInfo from '@/components/common/EthHashInfo'
+import { useEffect, useMemo } from 'react'
+import useSafeInfo from '@/hooks/useSafeInfo'
+import { useNestedSafeAddress, setNestedSafeAddress } from '@/components/common/WalletProvider'
+import TxCard from '@/components/tx-flow/common/TxCard'
+import InfoIcon from '@/public/images/notifications/info.svg'
+import SignatureIcon from '@/public/images/transactions/signature.svg'
+
+import css from './styles.module.css'
+import { sameAddress } from '@/utils/addresses'
+
+export const SignerForm = () => {
+  const wallet = useWallet()
+  const nestedSafeOwners = useNestedSafeOwners()
+  const nestedSafeAddress = useNestedSafeAddress()
+  const { safe } = useSafeInfo()
+
+  const onChange = (event: SelectChangeEvent<string>) => {
+    setNestedSafeAddress(event.target.value)
+  }
+
+  const isOwner = useMemo(
+    () => safe.owners.map((owner) => owner.value).includes(wallet?.address ?? ''),
+    [wallet?.address, safe.owners],
+  )
+
+  const isNotNestedOwner = useMemo(() => nestedSafeOwners && nestedSafeOwners.length === 0, [nestedSafeOwners])
+
+  const options = useMemo(
+    () =>
+      wallet
+        ? [wallet.address, ...(nestedSafeOwners ?? [])].filter((address) =>
+            safe.owners.some((owner) => sameAddress(owner.value, address)),
+          )
+        : [],
+    [nestedSafeOwners, safe.owners, wallet],
+  )
+
+  useEffect(() => {
+    if (nestedSafeAddress) {
+      return
+    }
+    if (!isOwner && nestedSafeOwners && nestedSafeOwners.length > 0) {
+      setNestedSafeAddress(nestedSafeOwners[0])
+    }
+    if (isOwner && (!nestedSafeOwners || nestedSafeOwners.length === 0)) {
+      setNestedSafeAddress(undefined)
+    }
+  }, [isOwner, nestedSafeAddress, nestedSafeOwners])
```

Comment:
What do you think about moving this to the context that sets the `signer`? Afaiu, this sets the state of the nested Safe address only on the `SignerForm`.